app/models/ext_management_system.rb

```diff
@@ -482,15 +484,29 @@ def vm_count_by_state(state)
     vms.inject(0) { |t, vm| vm.power_state == state ? t + 1 : t }
   end
 
-  def total_vms_on;        vm_count_by_state("on");        end
+  def host_count_by_metric(metric)
+    total_metric = hosts.inject(0) { |t, host| t + host.send(metric) }
+    case metric
+    when "ram_size"
+      ApplicationController.helpers.number_to_human_size(total_metric * 1024 * 1024)
+    else
+      total_metric
+    end
+  end
+
+  def total_cores;         host_count_by_metric("total_vcpus") end
+
+  def total_memory;        host_count_by_metric("ram_size")    end
+
+  def total_vms_on;        vm_count_by_state("on");            end
 
-  def total_vms_off;       vm_count_by_state("off");       end
+  def total_vms_off;       vm_count_by_state("off");           end
```

Comment:
if possible, please remove the formatting only changes from this PR.
They tend to be the culprit of merge conflicts, and it distracts from your pr